src/main/kotlin/org/malachite/estella/services/ApplicationService.kt

```diff
@@ -15,43 +14,42 @@ import java.util.*
 @Service
 class ApplicationService(
     @Autowired private val applicationRepository: ApplicationRepository,
-    @Autowired private val offerRepository: OfferRepository,
-    @Autowired private val jobSeekerRepository: JobSeekerRepository,
+    @Autowired private val offerService: OfferService,
+    @Autowired private val jobSeekerService: JobSeekerService,
     @Autowired private val recruitmentProcessService: RecruitmentProcessService,
     @Autowired private val interviewService: InterviewService,
     @Autowired private val mailService: MailService
-) {
-    fun insertApplicationLoggedInUser(offerId: Int, jobSeeker: JobSeeker, applicationPayload: ApplicationLoggedInPayload): Application {
-        val offer = offerRepository.findById(offerId).get()
+) : EStellaService<Application>() {
+
+    override val throwable: Exception = ApplicationNotFoundException()
+
+    fun insertApplicationLoggedInUser(
+        offerId: Int,
+        jobSeeker: JobSeeker,
+        applicationPayload: ApplicationLoggedInPayload
+    ): Application {
+        val offer = offerService.getOffer(offerId)
         val stage = offer.recruitmentProcess?.stages?.getOrNull(0)
         return stage?.let {
-            val application = applicationRepository.save(
-                applicationPayload.toApplication(it, jobSeeker)
-            )
-            if (application.seekerFiles.isNotEmpty()) {
-                val updatedSeekerFiles = jobSeeker.files.plus(application.seekerFiles)
-                val updatedJobSeeker = jobSeeker.copy(files = updatedSeekerFiles)
-                jobSeekerRepository.save(updatedJobSeeker)
-            }
+            val application = applicationRepository.save(applicationPayload.toApplication(it, jobSeeker))
+            if (application.seekerFiles.isNotEmpty())
+                jobSeeker.files.plus(application.seekerFiles)
+                    .let { jobSeekerService.updateJobSeekerFiles(jobSeeker, it) }
```

Comment:
Pytanie, czy na pewno chcemy tak robić? Że jakikolwiek on plik wrzuci, to jeb do jego plików dodajemy. Biorąc pod uwagę, że teoretycznie skończyliśmy już posiadanie konta (a nie mamy żadnych ustawień związanych z plikami), to trochę 2/10, że mu będziemy każdy plik wrzucać. Co jeśli wrzuci CV na pracę w barze, a potem będzie chciał urzędnika i HR zobaczy obydwa CV, jedno w którym pisze, że świetnie zabawia na imprezach, a w drugim, że jest poważną osobą XD

Reply:
Ponadto tu widzę jeszcze drugi problem - aplikuję na 10 ofert, do każdej dorzucam CV - czy update ogarnia to, że to jest dokładnie ten sam plik?

Reply:
1. To z tego co ogarnąłem to do aplikacji leci grupa plików ale podanych przez użytkownika więc do korpo będzie tylko to CV które chciał do korpo, ale wszystkie zaaplikowane CV miały się zapisywać w jego plikach by nie było duplikacji plików
2. Nie mam pojęcia czy to obsługuje, ale docelowo powinniśmy zrobić żeby działało XDDD